### PR TITLE
Update code to fix the issues related to byte[] type entity fields

### PR DESCRIPTION
**Pull request by sahanHe**
## Purpose

Fixes:[4075](https://github.com/ballerina-platform/ballerina-standard-library/issues/4075), [4074](https://github.com/ballerina-platform/ballerina-standard-library/issues/4074)

## Checklist
- [x] Linked to an issue
- [ ] ~Updated the specification~
- [x] Updated the changelog
- [x] Added tests

### CODE REVIEW

File: persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java [hunk 1]
```diff
@@ -205,6 +208,16 @@ public static void populateEntities(Module.Builder moduleBuilder, SyntaxTree bal
                     if (fieldNode.typeName().kind().equals(SyntaxKind.ARRAY_TYPE_DESC)) {
                         type = ((ArrayTypeDescriptorNode) fieldNode.typeName()).memberTypeDesc();
                         fieldBuilder.setArrayType(true);
+                    } else if (fieldNode.typeName().kind().equals(SyntaxKind.OPTIONAL_TYPE_DESC)) {
+                        if (((OptionalTypeDescriptorNode) fieldNode.typeName()).typeDescriptor().kind()
```
Comment (niveathika) [inline persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java:212]: This seems to repeat multiple times. Let move the optional or array check to the top of the if else check. 
Also `SyntaxKind.RECORD_FIELD_WITH_DEFAULT_VALUE` is not supported now.

File: persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java [hunk 2]
```diff
@@ -283,6 +296,8 @@ public static void inferRelationDetails(Module entityModule) {
                                             // one-to-many relation. associated entity is the owner.
                                             field.setRelation(computeRelation(entity, assocEntity, false));
                                             assocfield.setRelation(computeRelation(assocEntity, entity, true));
+                                        } else if (field.isArrayType() || field.getFieldType().equals("byte")) {
+                                            field.setRelation(null);
```
Comment (niveathika) [inline persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java:300]: Aren't the supported types defined as constants

Comment (sahanHe) [reply persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java:300]: The issue here was, when the field is array type, it was classified as a relation type. This is not true for byte arrays. So we have to specifically set it to null here as null is the default value for non related fields

File: persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java (hunk 1 above)
Comment (sahanHe) [reply persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java]: this was updated in [bebd8e7](https://github.com/ballerina-platform/persist-tools/pull/145/commits/bebd8e733cc86a2a4b5b1b71e4c55f6040f00d41) and [4e4b08a](https://github.com/ballerina-platform/persist-tools/pull/145/commits/4e4b08a8c2ce8225a924f387836bedef241805d0)

File: persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java [hunk 3]
```diff
@@ -176,48 +178,35 @@ public static void populateEntities(Module.Builder moduleBuilder, SyntaxTree bal
...
-                String qualifiedNamePrefix = null;
-                if (node.kind() == SyntaxKind.RECORD_FIELD_WITH_DEFAULT_VALUE) {
-                    RecordFieldWithDefaultValueNode fieldNode = (RecordFieldWithDefaultValueNode) node;
-
-                    fieldBuilder = EntityField.newBuilder(fieldNode.fieldName().text().trim());
-                    String fType;
-                    TypeDescriptorNode type;
-                    if (fieldNode.typeName().kind().equals(SyntaxKind.ARRAY_TYPE_DESC)) {
-                        type = ((ArrayTypeDescriptorNode) fieldNode.typeName()).memberTypeDesc();
+                RecordFieldNode fieldNode = (RecordFieldNode) node;
```
Comment (niveathika) [inline persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java:181]: Cant this logic be improved similar to https://github.com/ballerina-platform/module-ballerina-persist/blob/main/compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/PersistModelDefinitionValidator.java#L235-L295

As far as I can see, this logic too is trying to access the type and process if it is optional or array type.

Comment (sahanHe) [reply persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java:181]: updated in [c7dda48](https://github.com/ballerina-platform/persist-tools/pull/145/commits/c7dda48fdf8cf74c92d55ac0b96df0c4b9d30471)

File: persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxConstants.java [hunk 4]
```diff
@@ -173,6 +173,8 @@ private InheritedTypeReferenceConstants() {
     public static final String CLOSE_BRACE = "}";
     public static final String OPEN_BRACKET = "[";
     public static final String CLOSE_BRACKET = "]";
+    public static final String ARRAY = "[]";
+    public static final String PUNCTUATION = "?";
```
[the remark this answers is not in the document]
Comment (sahanHe) [reply persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxConstants.java]: updated in [c7dda48](https://github.com/ballerina-platform/persist-tools/pull/145/commits/c7dda48fdf8cf74c92d55ac0b96df0c4b9d30471)

**Review by niveathika — APPROVED** [review]
LGTM

File: persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_24/persist/foo_db_script.sql [hunk 5]
```diff
@@ -0,0 +1,7 @@
+DROP TABLE IF EXISTS ByteTest;
+
+CREATE TABLE ByteTest (
+	id INT NOT NULL,
+	binary1 BINARY NOT NULL,
```
Comment (daneshk) [inline persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_24/persist/foo_db_script.sql:5]: why `binaryOptional` field is not in the script

**Review by daneshk — COMMENTED** [review]
Add a minor comment. Others LGTM

File: persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_24/persist/foo_db_script.sql (hunk 5 above)
Comment (sahanHe) [reply persist-cli-tests/src/test/resources/test-src/output/tool_test_generate_24/persist/foo_db_script.sql]: updated in [a82dfeb](https://github.com/ballerina-platform/persist-tools/pull/145/commits/a82dfeb7f4ac126ec1bdc229582b746bde2f1d6f)

**Review by daneshk — APPROVED** [review]

**Pull request merged**